hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -71,15 +68,10 @@ public void start() {
     }
 
     public void refreshPartitions() {
-        if (!updating.compareAndSet(false, true)) {
-            return;
-        }
         ClientExecutionService executionService = client.getClientExecutionService();
         try {
-            ICompletableFuture future = executionService.submit(new RefreshTask());
-            future.andThen(refreshTaskCallback);
+            executionService.submit(new RefreshTask());
```

Comment:
the RefreshTask is using the User Executor rather than the internal executor. Should that be the case?